sematic/log_reader.py

```diff
@@ -501,6 +507,9 @@ def line_stream_from_log_directory(
         ),
         reverse=reverse,
     )
+    if len(log_files) == 0:
+        yield from []
+        return
```

Comment:
Is this covered in a unit test? It's difficult to make out due to the size of the PR.